From 22d7557905534d9e1b39f7d2a6d2036a40bf0c4e Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Wed, 10 Aug 2016 11:19:44 +0800
Subject: [PATCH] Fix missing check for permission denied.

If the path to "ROOT_SYSCONFDIR/mke2fs.conf" has a permission denied problem,
then the get_dirlist() call will return EACCES. But the code in profile_init
will treat that as a fatal error and all executions will fail with:
      Couldn't init profile successfully (error: 13).

Upstream-Status: Pending

Written-by: Henrik Wallin <henrik.b.wallin@ericsson.com>

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>

---
 lib/support/profile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/support/profile.c b/lib/support/profile.c
index f54739e7..53ea68f1 100644
--- a/lib/support/profile.c
+++ b/lib/support/profile.c
@@ -335,7 +335,7 @@ profile_init(const char * const *files, profile_t *ret_profile)
 				*last = new_file;
 				last = &new_file->next;
 			}
-		} else if ((retval != ENOTDIR) &&
+		} else if ((retval != ENOTDIR) && (retval != EACCES) &&
 			   strcmp(*fs, default_filename))
 			goto errout;
 
